plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/ui/contentassist/CustomCompletionProposalComputer.java

```diff
@@ -0,0 +1,97 @@
+package de.cognicrypt.codegenerator.ui.contentassist;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.internal.ui.text.java.JavaCompletionProposalComputer;
+import org.eclipse.jdt.ui.text.java.ContentAssistInvocationContext;
+import org.eclipse.jdt.ui.text.java.JavaContentAssistInvocationContext;
+import org.eclipse.jface.text.contentassist.CompletionProposal;
+import org.eclipse.jface.text.contentassist.ICompletionProposal;
+
+import de.cognicrypt.codegenerator.Activator;
+import de.cognicrypt.core.Constants;
+import de.cognicrypt.utils.CrySLUtils;
+
+
+@SuppressWarnings("restriction")
+public class CustomCompletionProposalComputer extends JavaCompletionProposalComputer {
+	
+	static private String rulesDir = Constants.ECLIPSE_RULES_DIR + Constants.innerFileSeparator + Constants.Rules.JavaCryptographicArchitecture.toString() + Constants.innerFileSeparator + 
```

Comment:
Would it make sense to store this as a constant in the constant class? If not, why?

Reply:
You are right! Since both content assist and code generator both uses latest jca ruleset from eclipse path, it makes sense to move it to Constants class